homeassistant/components/vegehub/config_flow.py

```diff
@@ -0,0 +1,260 @@
+"""Config flow for the VegeHub integration."""
+
+import logging
+from typing import Any
+
+from vegehub import VegeHub
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components import zeroconf
+from homeassistant.components.webhook import (
+    async_generate_id as webhook_generate_id,
+    async_generate_url as webhook_generate_url,
+)
+from homeassistant.config_entries import ConfigEntry, ConfigFlowResult
+from homeassistant.const import CONF_WEBHOOK_ID
+from homeassistant.core import callback
+
+from .const import DOMAIN, OPTION_DATA_TYPE_CHOICES
+
+_LOGGER = logging.getLogger(__name__)
+
+
+ip_dict: dict[str, str] = {}
+
+
+class VegeHubConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for VegeHub integration."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the VegeHub config flow."""
+        self._hub: VegeHub | None = None
+        self._hostname: str = ""
+        self._properties: dict = {}
+        self._config_url: str = ""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial confirmation step with no inputs."""
+        errors = {}
+
+        if user_input is not None:
+            if "ip_address" in user_input and self._hub is None:
+                # When the user has input the IP manually, we need to gather more information
+                # from the Hub before we can continue setup.
+                self._hub = VegeHub(str(user_input.get("ip_address")))
+
+                await self._hub.retrieve_mac_address()
+
+                if len(self._hub.mac_address) <= 0:
+                    _LOGGER.error(
+                        "Failed to get device config from %s", self._hub.ip_address
+                    )
+                    return self.async_abort(reason="cannot_connect")
+
+                try:
+                    # Check to see if this MAC address is already in the list.
+                    entry = list(ip_dict.keys())[
+                        list(ip_dict.values()).index(self._hub.mac_address)
+                    ]
+                    # If the mac address is on the list, pop it so we can give it a new IP
+                    if entry:
+                        ip_dict.pop(entry)
+                except ValueError:
+                    pass
```

Comment:
why are we passing an exception here?

Reply:
When a ValueError gets raised here it just means that the device wasn't on the list, so we don't need to remove it. I added a comment to explain.